internal/service/tenant_test.go

```diff
@@ -114,3 +114,105 @@ func (s *TenantServiceSuite) TestGetTenantByID() {
 		})
 	}
 }
+
+func (s *TenantServiceSuite) TestGetTenantWithBillingDetails() {
+	// Create tenant with billing details
+	tenantWithBilling := &tenant.Tenant{
+		ID:   "tenant-with-billing",
+		Name: "Billing Tenant",
+		BillingDetails: tenant.BillingDetails{
+			Email:     "billing@example.com",
+			HelpEmail: "help@example.com",
+			Phone:     "+1-555-987-6543",
+			Address: tenant.Address{
+				Line1:      "456 Market Street",
+				Line2:      "Floor 3",
+				City:       "San Francisco",
+				State:      "CA",
+				PostalCode: "94105",
+				Country:    "US",
+			},
+		},
+	}
+	_ = s.tenantRepo.Create(s.ctx, tenantWithBilling)
```

Comment:
can check err here and do 
s.NoErr(err) to validate creation is not leading to errors